src/Glpi/Application/Middleware/Authentication.php

```diff
@@ -68,6 +70,24 @@ public function __invoke(Request $request, Response $response, $next)
     {
         //TODO: do!
         //die('You shall not pass!')
+
+        $route = $request->getAttribute('route');
+        if (!$route || in_array($route->getName(), ['login', 'slash', 'cron', 'asset', 'do-login', 'lost-password'])) {
+            return $next($request, $response);
+        }
+
+        //redirect to login page if used is not logged in
+        if (!\Session::getLoginUserID()) {
+            $arguments = $route->getArguments();
+            $get = $request->getQueryParams();
+            $arguments = $arguments + $get;
+            //store current path for redirection
+            $_SESSION['glpi_redirect'] = $this->router->pathFor($route->getName(), $arguments);
+
+            $this->flash->addMessage('error', __('Authentication required'));
+            return $response->withRedirect($this->router->pathFor('login'), 301);
```

Comment:
301 is for permanent redirects and may be cached by browsers.
```suggestion
            return $response->withRedirect($this->router->pathFor('login'), 302);
```